### PR TITLE
Commit / tag on failure

### PR DESCRIPTION
**Pull request by rskang**
Also shows names as well as IDs in the Build Data section

Code by @djl197 I just merged it onto a more current master

[Review on Reviewable](https://reviewable.io/reviews/jenkinsci/docker-plugin/132)

### CODE REVIEW

**Comment by KostyaSha** [comment]
It doesn't cover all use cases, job status can be also Unstable. Some hetero list with dropdown cases should resolve it more dynamic and modular instead hardcoded checkboxes.

**Comment by magnayn** [comment]
Yep - this would be good (something we've found we need too, where a unit test fails it'd still be nice to pull the container to figure out why).

**Comment by rskang** [comment]
Yep. Selecting a threshold or a multi-select box to cover all cases is a much better idea.

File: src/main/java/com/nirima/jenkins/plugins/docker/DockerJobProperty.java [hunk 1]
```diff
@@ -73,6 +81,7 @@ public boolean isApplicable(Class<? extends Job> jobType) {
         public DockerJobProperty newInstance(StaplerRequest sr, JSONObject formData) throws hudson.model.Descriptor.FormException {
             return new DockerJobProperty(
                     (Boolean)formData.get("tagOnCompletion"),
+                    (Boolean)formData.get("tagOnFailure"),
```
Comment (KostyaSha) [inline src/main/java/com/nirima/jenkins/plugins/docker/DockerJobProperty.java]: Why not use getBoolean method?

File: src/main/java/com/nirima/jenkins/plugins/docker/DockerSlave.java [hunk 2]
```diff
@@ -112,34 +114,49 @@ protected void _terminate(TaskListener listener) throws IOException, Interrupted
...
             return;
         }
 
+        String repoName = theRun.getParent().getDisplayName();
+        String tagName = theRun.getResult().toString() + "_" + theRun.getDisplayName();
+        // Make sure we remove the # from the tagName
+        tagName = tagName.replace("#", "");
+        String taggedName = repoName + ":" + tagName;
+        
+        LOGGER.log(Level.INFO, "Going to tag the image locally - REPOSITORY: " + repoName + ". TAG: " + tagName);
```
Comment (KostyaSha) [inline src/main/java/com/nirima/jenkins/plugins/docker/DockerSlave.java]: Slow logging, remove string concatenations wit object argument

File: src/main/java/com/nirima/jenkins/plugins/docker/DockerSlave.java [hunk 3]
```diff
@@ -112,34 +114,49 @@ protected void _terminate(TaskListener listener) throws IOException, Interrupted
...
+        // then we need to check the job status.
+        if (!getJobProperty().isTagOnCompletion() &&
+            (getJobProperty().isTagOnFailure() && theRun.getResult() == Result.SUCCESS)) {
+            LOGGER.log(Level.INFO, "No need to tag the image as the build was SUCCESSFUL.");
+            addJenkinsAction(null, null);
             return;
         }
 
+        String repoName = theRun.getParent().getDisplayName();
+        String tagName = theRun.getResult().toString() + "_" + theRun.getDisplayName();
```
Comment (KostyaSha) [inline src/main/java/com/nirima/jenkins/plugins/docker/DockerSlave.java]: Why result is used for tag name?

**Comment by KostyaSha** [comment]
#181 related

**Comment by pjdarton** [comment]
Closing due to inactivity.

**Pull request closed without merging**